### ISSUE TITLE
Stripe sync not working

### ISSUE THREAD

**Issue by coffeeburrito** [service-bot/servicebot#80]
I'm testing ServiceBot for the first time, and did a fresh copy of the servicebot docker-compose yesterday. It tested the Stripe API keys and was able to import my Stripe users/plans/subscriptions etc.  At that time, it even created my ServiceBot user in Stripe as a customer. So far so good.

However, that's as much as I got.
1. Making changes in ServiceBot has no effect in Stripe. Adding/deleting/changing plans shows log entries like "service-templates 4 was updated by user...", but none of it gets propagated to Stripe.

2. Changes on Stripe do not appear in my ServiceBot instance. Stripe webhooks are getting rejected with the message "Bad Stripe Event!"
The logs show: servicebot_1  | POST /api/v1/stripe/webhook 404 421.747 ms - 17
I tested this using various events with Stripe's webhook tester and also by manually posting JSON to ServiceBot.

**Comment by darafsheh** [comment]
Hi @coffeeburrito . Thank you for trying out ServiceBot. I'll explain how ServiceBot integrates with Stripe and that should answer your questions:

1- Service Templates are ServiceBot specific entities that describe the schema of your service and Stripe payment plans. They do not create Stripe plans or subscription until the template is requested by the user. Once the Service Template is requested by a user/customer, ServiceBot will create the Stripe payment plan as well as the subscription, user fund, etc.

2- You can always start with you Stripe Test Mode API keys and run some request tests. Once you are ready, you can go to Stripe Settings page using your admin user account and change your keys to the Stripe Live keys, and ServiceBot will automatically move all your customers over to Stripe Live mode and your service templates will be ready to be requested on your ServiceBot site. Again, Service Templates won't show on your Stripe account until they are requested as services.

3- The webhook system validates the event_id that gets passed to it to ensure its validity. Servicebot currently supports the following webhook actions (we are adding more for the future releases):
- customer.deleted
- customer.subscription.deleted
- invoice.created
- invoice.updated
- invoice.payment_failed

Please let us know if you have any questions. We are also very interested to learn how you are using your ServiceBot and would certainly help you configure it if you need assistance.

Lastly, we would really appreciate a Star in our repo :)

**Comment by coffeeburrito** [comment]
Hi @darafsheh, thanks for the quick response. I am running in test mode with Stripe test keys. 

I see you are correct about the webhooks. I deleted a user in Stripe and that moved the user into a suspended state in ServiceBot. Thank you for that clarification.

As for the other issue, as a customer I attempt to sign up for a test service using a Stripe test card (I tried both VISA & mastercard). On submit, the page reloads and the credit card information is cleared. If I re-enter the credit card information, the submit button no longer works (I assume disabled to prevent accidental multiclicks)

No error is shown on the page, although I see the subscription request entity in the compose logs. There are still 0 services active. Stripe logs show that a card was created, but there was no followup. The docs suggest the next step would be to call the API and associate the card with the customer, but that doesn't seem to happen.

**Comment by darafsheh** [comment]
@coffeeburrito I wasn't able to reproduce that issue with our latest docker deployment. What version of ServiceBot are you running? You can check the package.json inside the container. To get inside the docker container:

First find the docker container id:
`docker ps | grep servicebotdeploy_servicebot_1`
Grab the container id, which should be the id listed on the first column.
Then run bash on that container:
`docker exec -it <CONTAINER ID> sh`
Once you're inside the container, look at the package.json file and let me know what is the version listed in there.

Also, did your initial setup completed successfully? or did you have to refresh the page for the initial loading to complete?

**Comment by coffeeburrito** [comment]
I went to screen record the issue today, but could no longer reproduce the problem. Everything seems to be working correctly now. On submit, instead of reloading the page it shows a spinny loading icon and then shows a subscription details page. Previously it had immediately reloaded the page without showing any loading icon at all. I guess it was a cache issue...? I'll mark the issue closed for now.

Package.json shows version 0.3.0. I seem to recall I did have to refresh the initial setup page, although I think it was because I forgot to set permissions on the docker volumes. I'm pretty sure I deleted and re-created since then.

As to your earlier question about how we plan to use ServiceBot: the smallish company where I work wants a self-service portal for customers to sign up, manage their subscription, and enter credit card info in a way that doesn't add PCI burdens. We were about to go live with a hosted solution when they increased their pricing to include a percent of revenue on top of Stripe's fees. ServiceBot seems to cover the basics
